src/tls/connection.c

```diff
@@ -429,28 +436,38 @@ static bool
 connection_connect_to_dynamic_wsinstance (Connection *self)
 {
   struct sockaddr_un addr = { .sun_family = AF_UNIX };
-  char fingerprint[65];
+  char fingerprint[FINGERPRINT_LENGTH + 1];
 
   assert (self->tls != NULL);
 
   if (!get_peer_certificate_fingerprint (self->tls, fingerprint, sizeof fingerprint))
     return false;
 
   sockaddr_printf (&addr, "%s/https@%s.sock", parameters.wsinstance_sockdir, fingerprint);
-
   debug (CONNECTION, "Connecting dynamic https wsinstance %s:", addr.sun_path);
+
   /* fast path: the socket already exists, so we can just connect to it */
   if (connect (self->ws_fd, (struct sockaddr *) &addr, sizeof addr) == 0)
     return true;
 
-  debug (CONNECTION, "  -> failed.  Requesting activation.");
+  if (errno != ENOENT && errno != ECONNREFUSED)
+    warn ("connect(%s) failed on the first attempt", addr.sun_path);
+
+  debug (CONNECTION, "  -> failed (%m).  Requesting activation.");
   /* otherwise, ask for the instance to be started */
   if (!request_dynamic_wsinstance (fingerprint))
     return false;
 
   /* ... and try one more time. */
   debug (CONNECTION, "  -> trying again");
-  return connect (self->ws_fd, (struct sockaddr *) &addr, sizeof addr) == 0;
+  if (connect (self->ws_fd, (struct sockaddr *) &addr, sizeof addr) != 0)
+    {
+      warn ("connect(%s) failed on the second attempt", addr.sun_path);
```

Comment:
%m

Reply:
`warn()` does it for itself

Reply:
GitHub bug report: There's no emoji for :brown_paperbag or :shame_on_me